fv3core/testing/validation.py

```diff
@@ -27,16 +28,25 @@ class SelectivelyValidated:
         """
 
         def __init__(self, *args, **kwargs):
+
             self.wrapped = cls(*args, **kwargs)
-            origin, domain = origin_domain_func(self.wrapped)
-            self._validation_slice = tuple(
-                slice(start, start + n) for start, n in zip(origin, domain)
-            )
+            self._validation_slice = {}
+
+            for arg_name, func in name_to_origin_domain_function.items():
+                variable_origin, variable_domain = func(self.wrapped)
+
+                self._validation_slice[arg_name] = tuple(
+                    slice(start, start + n)
+                    for start, n in zip(variable_origin, variable_domain)
+                )
+
             self._all_argument_names = tuple(
                 inspect.getfullargspec(self.wrapped).args[1:]
             )
             assert "self" not in self._all_argument_names
-            self._selective_argument_names = selective_arg_names
+            self._selective_argument_names = tuple(
+                name_to_origin_domain_function.keys()
+            )
```

Comment:
I might as well write this as a code suggestion along with the other two (see suggestion in set_nans).
```suggestion
```